public/index.js

```diff
@@ -139,6 +141,68 @@ async function initMap()
         
         map.controls[google.maps.ControlPosition.BOTTOM_LEFT].push(toggleStopsButtonDiv);
         loading = false;
+
+        let toggleAlertsButton = document.createElement('button');
+        toggleAlertsButton.id = 'toggleAlertsButton';
+        toggleAlertsButton.textContent = 'View Alerts';
+        toggleAlertsButton.classList.add(CSS_CLASS_TOGGLE_STOPS_BUTTON);
+        toggleAlertsButton.addEventListener('click', ToggleAlerts);
+        countdownDiv.appendChild(toggleAlertsButton);
+
+        let alertsDiv = document.createElement('div');
+        alertsDiv.id = 'alertsDiv';
+        alertsDiv.classList.add(CSS_CLASS_ALERTSDIV);
+
+        let alertsCloseButton = document.createElement('button');
+        alertsCloseButton.id = 'alertsCloseButton';
+        alertsCloseButton.textContent = 'X';
+        alertsCloseButton.addEventListener('click', ToggleAlerts);
+        alertsCloseButton.classList.add(CSS_CLASS_ALERTS_CLOSE_BUTTON);
+        alertsDiv.appendChild(alertsCloseButton);
+        alertsDiv.hidden = true;
+        
+        response = await fetch('alerts');
+        if (response.ok)
+        {
+            let alerts = await response.json();
+            alerts.forEach(alert =>
+            {
+                let alertType = document.createElement('h1');
+                alertType.innerText =  alert.alertType + ":";
+                alertsDiv.appendChild(alertType);
+                alertsDiv.appendChild(document.createElement('br'));
+
+                let alertText= document.createElement('h2');
+                alertText.innerText =  alert.descriptionText;
+                alertsDiv.appendChild(alertText);
+                alertsDiv.appendChild(document.createElement('br'));
+
+                let alertActivePeriod = document.createElement('p');
+                alertActivePeriod.innerText =  "Active from: " + alert.activePeriod.start + " To: " + alert.activePeriod.end;
+                alertsDiv.appendChild(alertActivePeriod);
+                alertsDiv.appendChild(document.createElement('br'));
+                alertsDiv.appendChild(document.createElement('br'));
+
+                let affectedRoutes = document.createElement('h4');
+                affectedRoutes.innerText =  "Affected Routes / Stops:";
+                alertsDiv.appendChild(affectedRoutes);
+                alertsDiv.appendChild(document.createElement('br'));
+
+                alert.affectedRoutes.forEach(stop =>

```

Comment:
We should rename the variable before the => to `route` as we're iterating through a list of routes